tokens/src/impls.rs

```diff
@@ -0,0 +1,209 @@
+	B: fungibles::Inspect<AccountId>,
+{
+	type AssetId = <B as fungibles::Inspect<AccountId>>::AssetId;
+	type Balance = <B as fungibles::Inspect<AccountId>>::Balance;
+
+	fn total_issuance(asset: Self::AssetId) -> Self::Balance {
+		if TestKey::contains(&asset) {
+			A::total_issuance()
+		} else {
+			B::total_issuance(asset)
+		}
+	}
+
+	fn minimum_balance(asset: Self::AssetId) -> Self::Balance {
+		if TestKey::contains(&asset) {
+			A::minimum_balance()
+		} else {
+			B::minimum_balance(asset)
+		}
+	}
+
+	fn balance(asset: Self::AssetId, who: &AccountId) -> Self::Balance {
+		if TestKey::contains(&asset) {
+			A::balance(who)
+		} else {
+			B::balance(asset, who)
+		}
+	}
+
+	fn reducible_balance(asset: Self::AssetId, who: &AccountId, keep_alive: bool) -> Self::Balance {
+		if TestKey::contains(&asset) {
+			A::reducible_balance(who, keep_alive)
+		} else {
+			B::reducible_balance(asset, who, keep_alive)
+		}
+	}
+
+	fn can_deposit(asset: Self::AssetId, who: &AccountId, amount: Self::Balance) -> DepositConsequence {
+		if TestKey::contains(&asset) {
+			A::can_deposit(who, amount)
+		} else {
+			B::can_deposit(asset, who, amount)
+		}
+	}
+
+	fn can_withdraw(
+		asset: Self::AssetId,
+		who: &AccountId,
+		amount: Self::Balance,
+	) -> WithdrawConsequence<Self::Balance> {
+		if TestKey::contains(&asset) {
+			A::can_withdraw(who, amount)
+		} else {
+			B::can_withdraw(asset, who, amount)
+		}
+	}
+}
+
+impl<AccountId, TestKey, A, B> fungibles::Transfer<AccountId> for Combiner<AccountId, TestKey, A, B>
+where
+	TestKey: Contains<<B as fungibles::Inspect<AccountId>>::AssetId>,
+	A: fungible::Transfer<AccountId, Balance = <B as fungibles::Inspect<AccountId>>::Balance>,
+	B: fungibles::Transfer<AccountId>,
+{
+	fn transfer(
+		asset: Self::AssetId,
+		source: &AccountId,
+		dest: &AccountId,
+		amount: Self::Balance,
+		keep_alive: bool,
+	) -> Result<Self::Balance, DispatchError> {
+		if TestKey::contains(&asset) {
+			A::transfer(source, dest, amount, keep_alive)
+		} else {
+			B::transfer(asset, source, dest, amount, keep_alive)
+		}
+	}
+}
+
+impl<AccountId, TestKey, A, B> fungibles::Mutate<AccountId> for Combiner<AccountId, TestKey, A, B>
+where
+	TestKey: Contains<<B as fungibles::Inspect<AccountId>>::AssetId>,
+	A: fungible::Mutate<AccountId, Balance = <B as fungibles::Inspect<AccountId>>::Balance>,
+	B: fungibles::Mutate<AccountId>,
+{
+	fn mint_into(asset: Self::AssetId, dest: &AccountId, amount: Self::Balance) -> DispatchResult {
+		if TestKey::contains(&asset) {
+			A::mint_into(dest, amount)
+		} else {
+			B::mint_into(asset, dest, amount)
+		}
+	}
+
+	fn burn_from(
+		asset: Self::AssetId,
+		dest: &AccountId,
+		amount: Self::Balance,
+	) -> Result<Self::Balance, DispatchError> {
+		if TestKey::contains(&asset) {
+			A::burn_from(dest, amount)
+		} else {
+			B::burn_from(asset, dest, amount)
+		}
+	}
+}
+
+pub trait ConvertBalance<A, B> {
+	fn convert_balance(amount: A) -> B;
```

Comment:
maybe we should just add currency id to here so this can be used to handle multiple rebase tokens.

Reply:
done